server/snapshot/snapshot.go

```diff
@@ -0,0 +1,161 @@
+package snapshot
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/Sirupsen/logrus"
+
+	"github.com/docker/notary/server/storage"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/signed"
+)
+
+// GetOrCreateSnapshotKey either creates a new snapshot key, or returns
+// the existing one. Only the PublicKey is returned. The private part
+// is held by the CryptoService.
+func GetOrCreateSnapshotKey(gun string, store storage.KeyStore, crypto signed.CryptoService, createAlgorithm string) (data.PublicKey, error) {
```

Comment:
Yeah, agreed. I basically copy/pasted the entire timestamp package and modified it to be snapshot. I wasn't sure initially how much difference there would be... Will probably fix once end to end PoC for server snapshotting is complete. I figure at that point there will be a bunch of refactoring we want to do.